### PR TITLE
Update mac.md for gcc instructions

### LINKED ISSUE #641: Mac OS X build instructions
**Describe the bug**
Failed to complete a HELICS build on Mac OS X 10.13.6 with gcc 7.3 and Python3 support by following the directions on RTD. The main problems were:

- Cmake looked for SWIG, and I gave up trying to build SWIG and its dependency, PCRE
- Make couldn't find Boost symbols for the x86_64 architecture, even though otool showed those to be present in the Boost libraries

**Additional context and information**
The following changes worked for me:

- Send -DENABLE_SWIG=OFF to Cmake.  The documentation suggests -DDISABLE_SWIG=ON (https://helics.readthedocs.io/en/latest/installation/language.html#helics-with-matlab) but that doesn't seem to work.  Please correct, and also add here: https://helics.readthedocs.io/en/latest/installation/mac.html#building-helics-using-gcc-and-python
- The documentation says Cmake >= 3.4 is required, and I already had 3.6.1, but apparently Cmake >= 3.8 is required to properly detect Boost 1.64. Cmake 3.14 is currently the latest version. Please instruct the user to "brew upgrade cmake" or download from cmake.org, if they already have Cmake installed.
- To finally build and test Boost, these instructions are what worked for me, substituting gcc 7.3 for 8.1: https://solarianprogrammer.com/2018/08/07/compiling-boost-gcc-clang-macos/
- To finally build and test HELICS, this is what worked for me: `cmake -DCMAKE_INSTALL_PREFIX="/Users/mcde601/local/helics-gcc-2.0.0/" -DBOOST_ROOT="/usr/local/boost-gcc-1.64" -DBUILD_PYTHON_INTERFACE=ON -DENABLE_SWIG=OFF -DUSE_BOOST_STATIC_LIBS=ON -DCMAKE_C_COMPILER=/usr/local/bin/gcc-7 -DCMAKE_CXX_COMPILER=/usr/local/bin/g++-7 ../` followed by `make clean; make -j 4; make install`

### PR DESCRIPTION
**Pull request by kdheepak**
Resolves #641

### CODE REVIEW

File: docs/installation/mac.md [hunk 1]
```diff
@@ -123,28 +123,49 @@ Out[2]: 'x.x.x (XX-XX-XX)'
 Building HELICS using gcc and python
 ------------------------------------
 
+Firstly, you'll need gcc. You can `brew install gcc`. Depending on the version of gcc you'll need to modify the following instructions slightly. These instructions are for `gcc-8.2.0`.
+
 First you will need to build boost using gcc from source. Download
```
Comment (phlptp) [inline docs/installation/mac.md]:
```suggestion
First you will need to build boost using gcc from source. Download the latest version from
```

File: docs/installation/mac.md [hunk 2]
```diff
@@ -123,28 +123,49 @@ Out[2]: 'x.x.x (XX-XX-XX)'
...
 
-This will install boost in the \~/local/boost-gcc-1.64 folder
+This will install boost in the `~/local/boost-gcc-1.69.0` folder
 
-Next, you will need to build HELICS and tell it what the BOOST\_ROOT is.
+Next, you will need to build HELICS and tell it what the `BOOST_ROOT` is.
 
 ```bash
-$ cmake -DCMAKE_INSTALL_PREFIX="/Users/$USER/local/helics-gcc-X.X.X/" -DBOOST_ROOT="/Users/$USER/local/boost-gcc-1.64" -DBUILD_PYTHON_INTERFACE=ON -DCMAKE_C_COMPILER=/usr/local/Cellar/gcc/7.2.0_1/bin/gcc-7 -DCMAKE_CXX_COMPILER=/usr/local/Cellar/gcc/7.2.0_1/bin/g++-7 ../
+$ cmake -DCMAKE_INSTALL_PREFIX="/Users/$USER/local/helics-gcc-X.X.X/" -DBOOST_ROOT="/Users/$USER/local/boost-gcc-1.64" -DBUILD_PYTHON_INTERFACE=ON -DCMAKE_C_COMPILER=/usr/local/Cellar/gcc/8.2.0/bin/gcc-8 -DCMAKE_CXX_COMPILER=/usr/local/Cellar/gcc/8.2.0/bin/g++-8 ../
```
Comment (phlptp) [inline docs/installation/mac.md]:
```suggestion
$ cmake -DCMAKE_INSTALL_PREFIX="/Users/$USER/local/helics-gcc-X.X.X/" -DBOOST_ROOT="/Users/$USER/local/boost-gcc-1.69" -DBUILD_PYTHON_INTERFACE=ON -DCMAKE_C_COMPILER=/usr/local/Cellar/gcc/8.2.0/bin/gcc-8 -DCMAKE_CXX_COMPILER=/usr/local/Cellar/gcc/8.2.0/bin/g++-8 ../
```

File: docs/installation/mac.md [hunk 3]
```diff
@@ -123,28 +123,49 @@ Out[2]: 'x.x.x (XX-XX-XX)'
 Building HELICS using gcc and python
 ------------------------------------
 
+Firstly, you'll need gcc. You can `brew install gcc`. Depending on the version of gcc you'll need to modify the following instructions slightly. These instructions are for `gcc-8.2.0`.
+
 First you will need to build boost using gcc from source. Download
-[boost](http://www.boost.org/users/history/version_1_64_0.html) from the
+[boost](http://www.boost.org/users/history/version_1_69_0.html) from the
 boost.org website.
```
Comment (phlptp) [inline docs/installation/mac.md]:
```suggestion
boost.org website.  Though keep in mind that your cmake version should be newer than the boost version, so if you have an older cmake you may want an older boost version.  
```

**Comment by phlptp** [comment]
May want to update the comments in the other similar instruction pages about cmake being newer than the boost version.   Also there was the comment in the issue about the SWIG_ENABLE section, which should be updated as well.

File: docs/installation/mac.md (hunk 3 above)
Comment (kdheepak) [reply docs/installation/mac.md]: Resolved in 4c67f3b

File: docs/installation/mac.md (hunk 2 above)
Comment (kdheepak) [reply docs/installation/mac.md]: Resolved in 4c67f3b

File: docs/installation/mac.md (hunk 1 above)
Comment (kdheepak) [reply docs/installation/mac.md]: Resolved in 4c67f3b

**Comment by phlptp** [comment]
@kdheepak  since you already have a PR can you update the link in the docs button on the readme to point to the readthedocs?

**Comment by phlptp** [comment]
And the update seems to have worked a lot better with the doxygen docs, now the correct mainpage is showing.

**Pull request merged**